File: src/main/java/infinitespire/relics/BeetleShell.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.common.*;
import com.megacrit.cardcrawl.actions.utility.UseCardAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.relics.AbstractRelic;

import infinitespire.abstracts.Relic;
import infinitespire.powers.BeetleShellPower;

public class BeetleShell extends Relic {

	public static final String ID = "Beetle Shell";
	
	public BeetleShell() {
		super(ID, "beetleshell", RelicTier.COMMON, LandingSound.SOLID);
		counter = 0;
	}
	
	@Override
	public AbstractRelic makeCopy() {
		return new BeetleShell();
	}
	
	@Override
    public void onUseCard(final AbstractCard card, final UseCardAction action) {
        if (card.baseBlock > 0) {
            ++this.counter;
            if (this.counter == 10) {
                this.counter = 0;
                this.flash();
                this.pulse = false;
            }
            
            if (this.counter == 9) {
                this.beginPulse();
                this.pulse = true;
                AbstractDungeon.player.hand.refreshHandLayout();
                AbstractDungeon.actionManager.addToBottom(new RelicAboveCreatureAction(AbstractDungeon.player, this));
                AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(AbstractDungeon.player, AbstractDungeon.player, new BeetleShellPower(AbstractDungeon.player), 1, true));
            }
        }
    }

	@Override
	public void atBattleStart() {
		AbstractPlayer p = AbstractDungeon.player;
		if (this.counter == 9) {
            this.beginPulse();
            this.pulse = true;
            AbstractDungeon.player.hand.refreshHandLayout();
			AbstractDungeon.actionManager.addToTop(new ApplyPowerAction(p,p, new BeetleShellPower(p, true)));
		}
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/BeetleShell.kt
package com.blanktheevil.infinitespire.relics


import com.blanktheevil.infinitespire.extensions.actionManager
import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.powers.BeetleShellPower
import com.blanktheevil.infinitespire.relics.abstracts.Relic
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction
import com.megacrit.cardcrawl.actions.utility.UseCardAction
import com.megacrit.cardcrawl.cards.AbstractCard

class BeetleShell : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "Beetle Shell".makeID()
    private const val IMG = "beetleshell"
    private val TIER = RelicTier.COMMON
    private val SOUND = LandingSound.SOLID
  }

  override fun onUseCard(card: AbstractCard, action: UseCardAction) {
    if (card.baseBlock > 0) {
      counter++
      if (counter == 10) {
        counter = 0
        flash()
        pulse = false
      }

      if (counter == 9) {
        beginPulse()
        pulse = true
        player.hand.refreshHandLayout()
        actionManager.addToBottom(
          RelicAboveCreatureAction(player, this)
        )
        player.applyPower(
          BeetleShellPower(player)
        )
      }
    }
  }

  override fun atBattleStart() {
    if (counter == 9) {
      beginPulse()
      pulse = true
      player.hand.refreshHandLayout()
      player.applyPower(
        BeetleShellPower(player)
      )
    }
  }
}